### PR TITLE
Review requirements - Authenticate a list of usernames

### PR DESCRIPTION
**Pull request by Paola-Xaq**
Hi Team, there is the assignment for the Reviewers.

### CODE REVIEW

File: src/main/java/org/fundacionjala/at13/authenticator/Auth.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
+package org.fundacionjala.at13;
+
+import java.util.Scanner;
+
+public class Auth {
+
+    public static void main(String[] args) {
+
+        int n = 0;
```
Comment (Raymundogs) [inline src/main/java/org/fundacionjala/at13/authenticator/Auth.java:9]: The variable name is not representative

Comment (GustavoZHA) [reply src/main/java/org/fundacionjala/at13/authenticator/Auth.java:9]: For example: numberUserName.

File: src/main/java/org/fundacionjala/at13/authenticator/Auth.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
+package org.fundacionjala.at13;
+
+import java.util.Scanner;
+
+public class Auth {
+
+    public static void main(String[] args) {
+
```
Comment (Raymundogs) [inline src/main/java/org/fundacionjala/at13/authenticator/Auth.java:8]: Unnecesary newline

File: src/main/java/org/fundacionjala/at13/authenticator/Auth.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        Scanner s1 = new Scanner(System.in);
+        Scanner s2 = new Scanner(System.in);
+        System.out.print("Enter no. of elements you want in array:");
+        n = s1.nextInt();
+        String usernames[] = new String[n];
+        System.out.println("Enter all the usernames:");
+        for (int i = 0; i < n; i++) {
+            usernames[i] = s2.nextLine();
+        }
+
```
Comment (Raymundogs) [inline src/main/java/org/fundacionjala/at13/authenticator/Auth.java:19]: Unnecesary newline

File: src/main/java/org/fundacionjala/at13/authenticator/Auth.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        System.out.println(validation(usernames));
+    }
+
+    // validation
```
Comment (Raymundogs) [inline src/main/java/org/fundacionjala/at13/authenticator/Auth.java:23]: The description is not clear on what the method does. It is to general.

File: src/main/java/org/fundacionjala/at13/authenticator/Auth.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    public static boolean validation(String[] array) {
+        for (int i = 0; i < array.length; i++) {
+            if ((sizeUsername(array[i]) && itHasLowerCase(array[i]) && itHasNumber(array[i]) && itHasOnlyNumberAndLowerCaseLetter(array[i]))==false) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    // it is between 6-10 characters long
```
Comment (Raymundogs) [inline src/main/java/org/fundacionjala/at13/authenticator/Auth.java:34]: On the description some ends in dot, others end with semi colon, it should be the same for all.

File: src/main/java/org/fundacionjala/at13/authenticator/Auth.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+            usernames[i] = s2.nextLine();
+        }
+
+        System.out.println(validation(usernames));
+    }
+
+    // validation
+
+    public static boolean validation(String[] array) {
+        for (int i = 0; i < array.length; i++) {
```
Comment (J0c4) [inline src/main/java/org/fundacionjala/at13/authenticator/Auth.java:26]: You can use foreach here

File: src/main/java/org/fundacionjala/at13/authenticator/Auth.java [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+            if ((sizeUsername(array[i]) && itHasLowerCase(array[i]) && itHasNumber(array[i]) && itHasOnlyNumberAndLowerCaseLetter(array[i]))==false) {
```
Comment (J0c4) [inline src/main/java/org/fundacionjala/at13/authenticator/Auth.java:27]: You can remove ==false

File: src/main/java/org/fundacionjala/at13/authenticator/Auth.java [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+            }
+        }
+        return false;
+    }
+
+    // contains at least 1 number;
+
+    public static boolean itHasNumber(String username) {
+        for (int i = 0; i < username.length(); i++) {
+            if ((int) username.charAt(i) >= 48 && (int) username.charAt(i) <= 57) {
```
Comment (J0c4) [inline src/main/java/org/fundacionjala/at13/authenticator/Auth.java:58]: magic number, please create a constant for those

File: src/main/java/org/fundacionjala/at13/authenticator/Auth.java [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+            }
+        }
+        return false;
+    }
+
+    // contains only numbers and lowercase letters.
+
+    public static boolean itHasOnlyNumberAndLowerCaseLetter(String username) {
+        for (int i = 0; i < username.length(); i++) {
+            if ((int) username.charAt(i) < 48 || ((int) username.charAt(i) > 57 && ((int) username.charAt(i) < 97) || (int) username.charAt(i) > 122)) {
```
Comment (J0c4) [inline src/main/java/org/fundacionjala/at13/authenticator/Auth.java:69]: magic number

File: src/main/java/org/fundacionjala/at13/authenticator/Auth.java [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+                return true;
+            }
+        }
+        return false;
+    }
+
+    // contains only numbers and lowercase letters.
+
+    public static boolean itHasOnlyNumberAndLowerCaseLetter(String username) {
+        for (int i = 0; i < username.length(); i++) {
```
Comment (J0c4) [inline src/main/java/org/fundacionjala/at13/authenticator/Auth.java:68]: can we try this? https://stackoverflow.com/questions/2451650/how-do-i-apply-the-for-each-loop-to-every-character-in-a-string

File: src/main/java/org/fundacionjala/at13/authenticator/Auth.java [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+    public static boolean sizeUsername(String username) {
+        if (username.length() >= 6 && username.length() <= 10) {
+            return true;
+        }
+        return false;
+    }
+
+    // contains at least 1 lowercase letter;
+
+    public static boolean itHasLowerCase(String username) {
```
Comment (J0c4) [inline src/main/java/org/fundacionjala/at13/authenticator/Auth.java:45]: divide and conquer, great work!

**Review by Raymundogs — APPROVED** [review]

**Review by JorgeCaceres-Jala — APPROVED** [review]

**Review by edsonarios — COMMENTED** [review]
Change approved

**Pull request merged**